src/backend/sgx/enclave.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Enclave object
+
+use std::fs::{File, OpenOptions};
+use std::io::Result;
+
+pub const SGX_ENCLAVE_DEVICE: &str = "/dev/sgx_enclave";
+
+/// Wraps an enclave file descriptor.
+pub struct Enclave(File);
+
+impl From<Enclave> for File {
+    fn from(enclave: Enclave) -> File {
+        enclave.0
+    }
+}
+
+impl From<File> for Enclave {
+    fn from(file: File) -> Enclave {
+        Self(file)
+    }
+}
```

Comment:
So I can create an Enclave from any open File? What's the purpose of this wrapper type then?

Reply:
Just remove this?

Reply:
I did:
```
diff --git a/src/backend/sgx/enclave.rs b/src/backend/sgx/enclave.rs
index 64da9ec..80b2f3f 100644
--- a/src/backend/sgx/enclave.rs
+++ b/src/backend/sgx/enclave.rs
@@ -16,18 +16,12 @@ impl From<Enclave> for File {
     }
 }
 
-impl From<File> for Enclave {
-    fn from(file: File) -> Enclave {
-        Self(file)
-    }
-}
-
 impl Enclave {
     pub fn new() -> Result<Enclave> {
         OpenOptions::new()
             .read(true)
             .write(true)
             .open(SGX_ENCLAVE_DEVICE)
-            .map(Into::into)
+            .map(Self)
     }
 }
```